proxy/handler.go

```diff
@@ -70,7 +72,18 @@ func (s *handler) handler(srv interface{}, serverStream grpc.ServerStream) error
 	}
 
 	clientCtx, clientCancel := context.WithCancel(outgoingCtx)
-	// TODO(mwitkow): Add a `forwarded` header to metadata, https://en.wikipedia.org/wiki/X-Forwarded-For.
+	md, ok := metadata.FromIncomingContext(clientCtx)
+	if !ok {
+		return status.Error(codes.Internal, ":authority header has not been set")
+	}
+	authority, ok := md[":authority"]
```

Comment:
is there any way of connecting these keys back to the upstream definition?

Reply:
This seems incorrect. `x-forwarded-for` is the list of peers forwarding the request. the authority is the intended destination of the request.

Reply:
This looks strange to me as well - I'm not clear on how `x-forwarded-host` and `authority` are related.

If you want to record the caller, then `x-forwarded-for` seems like a good choice. X-forwarded-for is implemented in my fork: https://github.com/vgough/grpc-proxy/blob/master/proxy/handler.go